src/Library.qs

```diff
@@ -19,63 +19,59 @@
         DataSize : Int);
 
     //Internal QRam type for a single QRAM Bit (is composed to form a full memory block)
-    internal newtype SingleBitQRAM = (Lookup : ((Qubit[], Qubit) => Unit is Adj + Ctl), 
-        AddressSize : Int);
+    internal newtype SingleBitQRAM = (Lookup : ((Qubit[], Qubit[]) => Unit is Adj + Ctl));
```

Comment:
This really isn't a single bit qRAM anymore, any ideas for a better name?

Reply:
Since this is the most general case (multiple address bits in, and multiple value bits out) I half-seriously suggest LooqUpTable / LooqUpBit, or just the general qROM.

Reply:
#sorrynotsorry but yeah let's keep this discussion open, I'll file it as an issue.